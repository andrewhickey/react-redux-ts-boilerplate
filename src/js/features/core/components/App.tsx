import * as React from 'react'

export default () => <div id="appContainer" />
