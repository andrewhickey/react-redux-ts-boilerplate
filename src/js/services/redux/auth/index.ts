import { handleActions } from "redux-actions"
import { AuthState } from "./types"
import login from "./login"

// export const USER_EXPIRED = require('./constants').USER_EXPIRED;
// export const REDIRECT_SUCCESS = require('./constants').REDIRECT_SUCCESS;
// export const USER_FOUND = require('./constants').USER_FOUND;
// export const USER_NOT_FOUND = require('./constants').USER_NOT_FOUND;
// export const SILENT_RENEW_ERROR = require('./constants').SILENT_RENEW_ERROR;
// export const SESSION_TERMINATED = require('./constants').SESSION_TERMINATED;
// export const USER_EXPIRING = require('./constants').USER_EXPIRING;
// export const LOADING_USER = require('./constants').LOADING_USER;
// export const USER_SIGNED_OUT = require('./constants').USER_SIGNED_OUT;

const initialState: AuthState = {
  user: null,
  working: false,
  error: null
}

export const reducer = handleActions(
  {
    ...login.reducer
  },
  initialState
)
