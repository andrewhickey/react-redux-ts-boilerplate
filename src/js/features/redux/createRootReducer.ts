import { combineReducers } from "redux"
import { reducer as formReducer } from "redux-form"
import { reducer as authReducer } from "../auth"

export default () =>
  combineReducers({
    auth: authReducer,
    form: formReducer,
  })
