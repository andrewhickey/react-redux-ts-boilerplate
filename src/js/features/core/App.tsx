import * as React from "react"
import * as ReactDOM from "react-dom"
import { BrowserRouter as Router } from "react-router-dom"
import { Store } from "redux"
import { Provider } from "react-redux"

export interface AppProps {
  store: Store<any>
}
export default ({ store }: AppProps) => (
  <Provider store={store}>
    <div className="appContainer" />
  </Provider>
)
