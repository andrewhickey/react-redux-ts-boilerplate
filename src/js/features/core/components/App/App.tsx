import * as React from "react"
import * as ReactDOM from "react-dom"
import { Store } from "redux"

export default () => <div id="appContainer">app</div>
