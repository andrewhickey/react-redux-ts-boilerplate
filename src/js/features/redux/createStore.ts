import { applyMiddleware, createStore, Middleware, Reducer } from 'redux'
import * as promiseMiddleware from 'redux-promise'
import thunkMiddleware from 'redux-thunk'

export default (
  rootReducer: Reducer<{}>,
  startingState: any = {},
  middlewares: Middleware[] = []
) => {
  const middleware = applyMiddleware(
    promiseMiddleware,
    thunkMiddleware,
    ...middlewares
  )
  const store = createStore(rootReducer, startingState, middleware)
  return store
}
