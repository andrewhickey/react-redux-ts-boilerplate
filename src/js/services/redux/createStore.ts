import { createStore, applyMiddleware } from "redux"
import * as promiseMiddleware from "redux-promise"
import thunkMiddleware from "redux-thunk"
import rootReducer from "./rootReducer"

export default (startingState: any, authManager: any) => {
  const middleware = applyMiddleware(
    thunkMiddleware.withExtraArgument({ authManager }),
    promiseMiddleware
  )
  const store = createStore(rootReducer)
  return store
}
