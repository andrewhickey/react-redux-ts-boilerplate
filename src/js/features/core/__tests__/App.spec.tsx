import App, { AppProps } from "../App"
import { createStore } from "redux"
import "jest-enzyme"
import * as React from "react"
import * as ReactDOM from "react-dom"
import * as ReactTestRenderer from "react-test-renderer"
import { shallow } from "enzyme"

const testReducer = (state: string = "") => state
const store = createStore(testReducer)
const props: AppProps = { store }

it("matches the stored snapshot when rendering with a store", () => {
  const tree = ReactTestRenderer.create(<App {...props} />).toJSON()
  expect(tree).toMatchSnapshot()
})

it("contains an element with class appContainer", () => {
  const wrapper = shallow(<App {...props} />)
  expect(wrapper.find(".appContainer")).toBePresent()
})
