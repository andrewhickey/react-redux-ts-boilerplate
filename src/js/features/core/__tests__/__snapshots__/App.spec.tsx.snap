// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the stored snapshot 1`] = `
<div
  id="appContainer"
/>
`;
